src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,175 @@
+import logging
+import time
+from multiprocessing import Process
+from typing import Any, Dict, Optional
+
+import requests
+import torch
+from typing_extensions import Literal
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.serve.servable_module import ServableModule
+from pytorch_lightning.strategies import DDPFullyShardedNativeStrategy, DDPFullyShardedStrategy, DeepSpeedStrategy
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _RequirementAvailable
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
+
+_NOT_SUPPORTED_STRATEGIES = (
+    DeepSpeedStrategy,
+    DDPFullyShardedNativeStrategy,
+    DDPFullyShardedStrategy,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class ServableModuleValidator(Callback):
+    """The ServableModuleValidator validates to validate a model correctly implement the ServableModule API.
+
+    .. warning::
+
+        This is currently an experimental feature and API changes are to be expected.
+
+    Arguments:
+        optimization: The format in which the model should be tested while being served.
+        server: The library used to evaluate the model serving. Default is FastAPI.
+        host: The host address associated the server.
+        port: The port associated the server.
+    """
+
+    def __init__(
+        self,
+        optimization: Optional[Literal["trace", "script", "onnx", "tensorrt"]] = None,
+        server: Literal["fastapi", "ml_server", "torchserve", "sagemaker"] = "fastapi",
+        host: str = "127.0.0.1",
+        port: int = 8080,
+        timeout: int = 10,
+        exit_on_failure: bool = True,
+    ):
+        super().__init__()
+        fastapi_installed = _RequirementAvailable("fastapi")
+        if not fastapi_installed:
+            raise ModuleNotFoundError(fastapi_installed.message)
+        uvicorn_installed = _RequirementAvailable("uvicorn")
+        if not uvicorn_installed:
+            raise ModuleNotFoundError(uvicorn_installed.message)
+
+        # TODO: Add support for the other options
+        if optimization is not None:
+            raise NotImplementedError(f"The optimization {optimization} is currently not supported.")
+        # TODO: Add support for testing with those server services
+        if server != "fastapi":
+            raise NotImplementedError("Only the fastapi server is currently supported.")
+
+        self.optimization = optimization
+        self.host = host
+        self.port = port
+        self.server = server
+        self.timeout = timeout
+        self.exit_on_failure = exit_on_failure
+        self.resp: Optional[requests.Response] = None
+
+    @rank_zero_only
+    def on_train_start(self, trainer: "pl.Trainer", servable_module: "pl.LightningModule") -> None:
```

Comment:
should it be enabled with `fast_dev_run`?

Reply:
I think so too. This is quite fast.